lib/redis_sync_time_client.rb

```diff
@@ -0,0 +1,25 @@
+
+class RedisSyncTimeClient
+  KEY_EXPIRE = 60 * 60 * 24 * 7
```

Comment:
So now we will expire barcodes based on how long they have been in the queue instead of how big the queue is?

Reply:
Yes. When we were storing the lookup in the class, the risk was that it would grow too large and eat up all of the memory, so restricting by size made sense. Now that we're using a big redis server, we just want to make sure keys are deleted *eventually*. The keys are only useful to the extent they identify redundant updates. Redundant updates only happen when we have a backlog. A backlog exceeding 7 days feels unlikely, hence we delete these tiny redis entries after 7 days.